contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -0,0 +1,560 @@
+    uint32 gasLimit;
+    uint96 estimatedCost;
+    uint256 timeoutTimestamp;
+    uint256 gasAfterPaymentCalculation;
+    uint96 adminFee;
+  }
+
+  mapping(bytes32 => Request) /* request ID */ /* Request data */ internal s_requests;
+
+  struct Receipt {
+    uint96 callbackGasCostJuels;
+    uint96 totalCostJuels;
+  }
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(address link) {
+    LINK = LinkTokenInterface(link);
+  }
+
+  function getTotalBalance() external view returns (uint256) {
+    return s_totalBalance;
+  }
+
+  // ================================================================
+  // |                      Getter methods                          |
+  // ================================================================
+  function _isValidSubscription(uint64 subscriptionId) internal view {
+    if (s_subscriptions[subscriptionId].owner == address(0)) {
+      revert InvalidSubscription();
+    }
+  }
+
+  function _isValidConsumer(address client, uint64 subscriptionId) internal view {
+    if (s_consumers[client][subscriptionId].allowed == false) {
+      revert InvalidConsumer(subscriptionId, client);
+    }
+  }
+
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function getSubscription(
+    uint64 subscriptionId
+  )
+    external
+    view
+    returns (uint96 balance, uint96 blockedBalance, address owner, address requestedOwner, address[] memory consumers)
+  {
+    _isValidSubscription(subscriptionId);
+    return (
+      s_subscriptions[subscriptionId].balance,
+      s_subscriptions[subscriptionId].blockedBalance,
+      s_subscriptions[subscriptionId].owner,
+      s_subscriptions[subscriptionId].requestedOwner,
+      s_subscriptions[subscriptionId].consumers
+    );
+  }
+
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function getConsumer(
+    address client,
+    uint64 subscriptionId
+  ) external view returns (bool allowed, uint64 initiatedRequests, uint64 completedRequests) {
+    return (
+      s_consumers[client][subscriptionId].allowed,
+      s_consumers[client][subscriptionId].initiatedRequests,
+      s_consumers[client][subscriptionId].completedRequests
+    );
+  }
+
+  // ================================================================
+  // |                 Internal Payment methods                     |
+  // ================================================================
+  /**
+   * @notice Blocks funds on a subscription account to be used for an in flight request
+   * @dev Only callable within the Router
+   * @param client - the consumer contract that initiated the request
+   * @param subscriptionId - The subscription ID to block funds for
+   * @param amount - The amount to transfer
+   */
+  function _blockBalance(address client, uint64 subscriptionId, uint96 amount) internal {
```

Comment:
could this function name also somehow indicate that it also increments `initiatedRequests`?

Reply:
Any suggestions?
Could go verbose like `_blockBalanceAndCountRequest` or general like `_markInFlight`

Reply:
maybe its fine as is.  I'll resolve this comment.